### PR TITLE
[MULTIARCH-3975] New IBM Z LPAR installation assembly

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s): 4.15+

Issue: https://issues.redhat.com/browse/MULTIARCH-3975

Link to docs preview: 
- [Preparing to install on IBM Z and IBM LinuxONE](https://71071--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/preparing-to-install-on-ibm-z)
- [Installing a cluster in an LPAR on IBM Z® and IBM® LinuxONE](https://71071--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-lpar#installing-ibm-z-lpar)
- [Installing a cluster in an LPAR on IBM Z and IBM LinuxONE in a restricted network](https://71071--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z-lpar)

QE review:
- [x] QE has approved this change.

Additional information:
- [Gdoc - LPAR installation](https://docs.google.com/document/d/1pUmPuHgycKxeravWiuMMU3p613UGgREJXKqiB6CK9yc/edit#heading=h.595c088ryqc5)
- [Gdoc - Preparing to install](https://docs.google.com/document/d/1weJA9NBMk5fXMc6W7Xf65NF12xeD6W9NfUFpzznZVmM/edit)

### CODE REVIEW

File: modules/minimum-ibm-z-system-requirements.adoc [hunk 1]
```diff
@@ -12,6 +27,12 @@ You can install {product-title} version {product-version} on the following {ibm-
 * {ibm-name} z16 (all models), {ibm-name} z15 (all models), {ibm-name} z14 (all models)
 * {ibm-linuxone-name} 4 (all models), {ibm-linuxone-name} III (all models), {ibm-linuxone-name} Emperor II, {ibm-linuxone-name} Rockhopper II
 
+[WARNING]
+====
+Running {product-title} on {ibm-z-name} in processor resource/system manager (PR/SM) mode without a hypervisor is strongly discouraged. Instead, use the Dynamic Partition Manager (DPM).
```
Comment (mjpytlak) [inline modules/minimum-ibm-z-system-requirements.adoc:32]: Generally we should avoid recommendations, or in this case discouraging [1], instead explaining why something should or should not be down. I suspect that the KB you are waiting on might explain why. I suggest rewording and letting the KB do the heavy lifting here.

[1] https://redhat-documentation.github.io/supplementary-style-guide/#recommend

File: modules/preferred-ibm-z-system-requirements.adoc [hunk 2]
```diff
@@ -2,6 +2,21 @@
...
+endif::[]
+ifeval::["{context}" == "installing-restricted-networks-ibm-z"]
+:ibm-z:
+endif::[]
+ifeval::["{context}" == "installing-ibm-z-lpar"]
+:ibm-z-lpar:
+endif::[]
+ifeval::["{context}" == "installing-restricted-networks-ibm-z-lpar"]
+:ibm-z-lpar:
+endif::[]
```
Comment (mjpytlak) [inline modules/preferred-ibm-z-system-requirements.adoc:19]: You set `ibm-z-lpar` here and on line 14, but never use it. Does it make sense to replace instances of `ifndef::ibm-z[]` with `ifdef::ibm-z-lpar`?

File: modules/preferred-ibm-z-system-requirements.adoc [hunk 3]
```diff
@@ -12,11 +27,17 @@
 
 * Three LPARS that each have the equivalent of six IFLs, which are SMT2 enabled, for each cluster.
 * Two network connections to both connect to the `LoadBalancer` service and to serve data for traffic outside the cluster.
+ifdef::ibm-z[]
 * HiperSockets, which are attached to a node either directly as a device or by bridging with one z/VM VSWITCH to be transparent to the z/VM guest. To directly connect HiperSockets to a node, you must set up a gateway to the external network via a {op-system-base} 8 guest to bridge to the HiperSockets network.
+endif::ibm-z[]
+ifndef::ibm-z[]
+* HiperSockets, which are attached to a node directly as a device. To directly connect HiperSockets to a node, you must set up a gateway to the external network via a {op-system-base} 8 guest to bridge to the HiperSockets network.
```
Comment (mjpytlak) [inline modules/preferred-ibm-z-system-requirements.adoc:34]:
```suggestion
* HiperSockets that are attached to a node directly as a device. To directly connect HiperSockets to a node, you must set up a gateway to the external network via a {op-system-base} 8 guest to bridge to the HiperSockets network.
```

File: modules/preferred-ibm-z-system-requirements.adoc [hunk 4]
```diff
@@ -25,24 +46,61 @@ On your z/VM instances, set up:
...
+// === Disk storage for the z/VM guest virtual machines
+// === Disk storage for the LPARs
 
-* FICON attached disk storage (DASDs). These can be z/VM minidisks, fullpack minidisks, or dedicated DASDs, all of which must be formatted as CDL, which is the default. To reach the minimum required DASD size for {op-system-first} installations, you need extended address volumes (EAV). If available, use HyperPAV and High Performance FICON (zHPF) to ensure optimal performance.
+ifdef::ibm-z[]
+* FICON attached disk storage (DASDs). These can be z/VM minidisks, fullpack minidisks, or dedicated DASDs, all of which must be formatted as CDL, which is the default. To reach the minimum required DASD size for {op-system-first} installations, you need extended address volumes (EAV). If available, use HyperPAV to ensure optimal performance.
 * FCP attached disk storage
+endif::ibm-z[]
+ifndef::ibm-z[]
+* FICON attached disk storage (DASDs). This can be dedicated DASDs that must be formatted as CDL, which is the default. To reach the minimum required DASD size for {op-system-first} installations, you need extended address volumes (EAV). If available, use HyperPAV to ensure optimal performance.
```
Comment (mjpytlak) [inline modules/preferred-ibm-z-system-requirements.adoc:83]:
```suggestion
* FICON attached disk storage (DASDs). These can be dedicated DASDs that must be formatted as CDL, which is the default. To reach the minimum required DASD size for {op-system-first} installations, you need extended address volumes (EAV). If available, use HyperPAV to ensure optimal performance.
```

**Review by mjpytlak — COMMENTED** [review]
@SNiemann15 Feedback for your review. Thanks.

**Comment by mjpytlak** [comment]
If you have questions about the rebase @SNiemann15 please reach out. @JoeAldinger deleted several `ifeval` statements that he had mistakenly added to the file on a rebase. You were also deleting these `ifevals` in `modules/installation-load-balancing-user-infra.adoc`. Happy to walk through the rebase if it is not clear on what to do.

File: installing/installing_ibm_z/installing-ibm-z-lpar.adoc [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+For a cluster that contains user-provisioned infrastructure, you must deploy all
+of the required machines.
+
+This section describes the requirements for deploying {product-title} on user-provisioned infrastructure.
+
+include::modules/installation-machine-requirements.adoc[leveloffset=+2]
+include::modules/installation-minimum-resource-requirements.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+.Additional resources
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply installing/installing_ibm_z/installing-ibm-z-lpar.adoc:51]: Good to know then I'll remove the ids. I think they came in through a peer review some time back.

File: installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc [hunk 6]
```diff
@@ -23,26 +23,75 @@ While this document refers only to {ibm-z-name}, all information in it also appl
...
+* *Local Agent-based*: You can deploy a cluster locally with the xref:../../installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer.adoc#preparing-to-install-with-agent-based-installer[agent-based installer] for air-gapped or restricted networks. It provides many of the benefits of the {ai-full}, but you must download and configure the link:https://console.redhat.com/openshift/install/ibmz/agent-based[agent-based installer] first. Configuration is done with a command line interface (CLI). This approach is ideal for air-gapped or restricted networks.
+//agent-based link for IBM Z not available yet needs to be verified shortly before GA
+* *Full control*: You can deploy a cluster on xref:../../installing/installing_ibm_z/installing-ibm-z.adoc#installing-ibm-z[infrastructure that you prepare and maintain], which provides maximum customizability. You can deploy clusters with both connected or air-gapped or restricted networks.
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc:32]: I'll take the wording from this [page](https://docs.openshift.com/container-platform/4.14/installing/index.html): 
You can deploy clusters in connected or disconnected environments.

@mjpytlak Shall I pick up that change for bare metal in this file too? https://docs.openshift.com/container-platform/4.14/installing/installing_bare_metal/preparing-to-install-on-bare-metal.html#choosing-a-method-to-install-ocp-on-bare-metal

[the remark this answers is not in the document]
Comment (mjpytlak) [reply installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc:32]: @SNiemann15 In looking at this again, what you had previously is probably fine. Feel free to take/leave anything I leave behind for copy edits. In looking at this, I was mostly focused on items around conditions.

File: installing/installing_ibm_z/installing-ibm-z-lpar.adoc [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+* link:https://access.redhat.com/solutions/4387261[How to generate SOSREPORT within OpenShift4 nodes without SSH].
+
+[id="next-steps_ibmz-lpar"]
+== Next steps
+
+* xref:../../post_installation_configuration/machine-configuration-tasks.adoc#rhcos-enabling-multipath_post-install-machine-configuration-tasks[Enabling multipathing with kernel arguments on {op-system}].
+
+* xref:../../post_installation_configuration/cluster-tasks.adoc#available_cluster_customizations[Customize your cluster].
+
+* If necessary, you can
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply installing/installing_ibm_z/installing-ibm-z-lpar.adoc:154]: Wouldn't it be better to stay in sync with the other providers? My VSC search has 46 results in 28 files on `* If necessary, you can xref`

[the remark this answers is not in the document]
Comment (mjpytlak) [reply installing/installing_ibm_z/installing-ibm-z-lpar.adoc:154]: All I am saying is that our guidelines have us avoiding text in Additional resources. I am not sure if this is a Jupiter requirement that is now no longer relevant. Maybe take a question to the docs forum? I agree that text is required in some of the xrefs for context, but as a peer reviewer I do need to point out the guideline.

File: installing/installing_ibm_z/installing-restricted-networks-ibm-z-lpar.adoc [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+[role="_additional-resources"]
+.Additional resources
+
+* link:https://access.redhat.com/solutions/4387261[How to generate SOSREPORT within {product-title} version 4 nodes without SSH].
+
+[id="next-steps_ibmz-lpar-restricted"]
+== Next steps
+
+* xref:../../post_installation_configuration/cluster-tasks.adoc#available_cluster_customizations[Customize your cluster].
+* If the mirror registry that you used to install your cluster has a trusted CA, add it to the cluster by xref:../../openshift_images/image-configuration.adoc#images-configuration-cas_image-configuration[configuring additional trust stores].
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply installing/installing_ibm_z/installing-restricted-networks-ibm-z-lpar.adoc:155]: Same comment as before, shouldn't this rather be in sync with other providers? There's 12 results for this in next steps.

File: installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc [hunk 9]
```diff
@@ -23,26 +23,75 @@ While this document refers only to {ibm-z-name}, all information in it also appl
...
+* *Interactive*: You can deploy a cluster with the web-based link:https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform[{ai-full}]. This method requires no setup for the installer, and is ideal for connected environments like {ibm-z-name}.
 
-See the xref:../../architecture/architecture-installation.adoc#installation-process_architecture-installation[Installation process] for more information about Assisted Installer and user-provisioned installation processes.
+* *Local Agent-based*: You can deploy a cluster locally with the xref:../../installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer.adoc#preparing-to-install-with-agent-based-installer[agent-based installer] for air-gapped or restricted networks. It provides many of the benefits of the {ai-full}, but you must download and configure the link:https://console.redhat.com/openshift/install/ibmz/agent-based[agent-based installer] first. Configuration is done with a command line interface (CLI). This approach is ideal for air-gapped or restricted networks.
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc:30]: Yes correct it's Agent-base Installer. This should probably also be changed for bare metal @skopacz1

File: modules/installation-bare-metal-config-yaml.adoc [hunk 10]
```diff
@@ -72,12 +81,18 @@ endif::[]
 ifeval::["{context}" == "installing-ibm-z-kvm"]
   architecture: s390x
 endif::[]
+ifeval::["{context}" == "installing-ibm-z-lpar"]
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/installation-bare-metal-config-yaml.adoc:84]: If I remember correctly @vikram-redhat suggested to code like this some years back. I will look into this through a separate PR when this one is merged.

File: modules/installation-initializing-manual.adoc [hunk 11]
```diff
@@ -16,6 +16,11 @@
 // * installing/installing_vsphere/installing-vsphere.adoc
 // * installing/installing_vsphere/installing-vsphere-network-customizations.adoc
 // * installing/installing_ibm_z/installing-ibm-z.adoc
+// * installing/installing_ibm_z/installing-restricted-networks-ibm-z.adoc
+// * installing/installing_ibm_z/installing-ibm-z-kvm.adoc
+// * installing/installing_ibm_z/installing-restricted-networks-ibm-z-kvm.adoc
+// * installing/installing_ibm_z/installing-ibm-z-lpar.adoc
+// * installing/installing_ibm_z/installing-restricted-networks-ibm-z-lpar.adoc
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/installation-initializing-manual.adoc:23]: @mjpytlak Thanks for pointing that out. The output does look the same as in previous releases. However, I noticed that the abstract doesn't show also for other providers and back to OCP 4.11. 
I think changing lines 87-89 to: 
```
ifndef::vsphere-upi,restricted-upi[]
For user-provisioned installations of {product-title}, you manually generate your installation configuration file.
endif::vsphere-upi,restricted-upi[]
```
solves the issue however I can't backport my PR. 
Again something for a separate PR?

[the remark this answers is not in the document]
Comment (mjpytlak) [reply modules/installation-initializing-manual.adoc:23]: Yes. A separate PR would be fine. Thanks.

File: installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (skopacz1) [reply installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc:30]:
```suggestion
* *Local Agent-based*: You can deploy a cluster locally with the xref:../../installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer.adoc#preparing-to-install-with-agent-based-installer[Agent-based Installer] for air-gapped or restricted networks. It provides many of the benefits of the {ai-full}, but you must download and configure the link:https://console.redhat.com/openshift/install/ibmz/agent-based[Agent-based Installer] first. Configuration is done with a command line interface (CLI). This approach is ideal for air-gapped or restricted networks.
```

[the remark this answers is not in the document]
Comment (skopacz1) [reply installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc:30]: I suppose in this context the phrase "Local Agent-based" works, though I think this is one of the only times "Agent-based" is used as a standalone adjective.

File: modules/installation-bare-metal-config-yaml.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (kalexand-rh) [reply modules/installation-bare-metal-config-yaml.adoc:84]: @SNiemann15, the only times you use the ifeval statements are at the beginning and end of a module to set it up for varying content based on the assembly. The content changes themselves are enclosed in ifdef/ifndef statements. [Here](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#conditionals-for-different-assemblies) is the relevant part of the contrib guide.

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 12]
```diff
@@ -1,12 +1,34 @@
...
+endif::[]
+ifeval::["{context}" == "installing-restricted-networks-ibm-z"]
+:ibm-z:
+endif::[]
+ifeval::["{context}" == "installing-ibm-z-lpar"]
+:ibm-z-lpar:
+endif::[]
+ifeval::["{context}" == "installing-restricted-networks-ibm-z-lpar"]
+:ibm-z-lpar:
+endif::[]
```
[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/installation-ibm-z-user-infra-machines-iso.adoc:20]: Agree they both work I'll change.

File: modules/minimum-ibm-z-system-requirements.adoc (hunk 1 above)
Comment (SNiemann15) [reply modules/minimum-ibm-z-system-requirements.adoc:32]: Correct wording is under discussion. I'll change once we have an agreement.

File: installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (SNiemann15) [reply installing/installing_ibm_z/preparing-to-install-on-ibm-z.adoc]: I changed to the simpler phrasing those two "or" confused me too.

File: modules/installation-bare-metal-config-yaml.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/installation-bare-metal-config-yaml.adoc:84]: I will pick this up in a follow-up PR and backport to earlier releases.

**Comment by sandisamp** [comment]
LGTM

**Comment by holgwolf** [comment]
LGTM

**Comment by bscott-rh** [comment]
Added peer review done label per slack conversation with Mike.
